clientv3/client.go

```diff
@@ -213,16 +217,16 @@ func (c *Client) retryConnection(err error) (newConn *grpc.ClientConn, dialErr e
 	if err != nil {
 		c.errors = append(c.errors, err)
 	}
-	if c.cancel == nil {
-		return nil, c.ctx.Err()
-	}
 	if c.conn != nil {
 		c.conn.Close()
 		if st, _ := c.conn.State(); st != grpc.Shutdown {
 			// wait so grpc doesn't leak sleeping goroutines
-			c.conn.WaitForStateChange(c.ctx, st)
+			c.conn.WaitForStateChange(context.Background(), st)
 		}
 	}
+	if c.cancel == nil {
```

Comment:
I feel we need to add a comment about this check. cancel does not tell the client is closed explicitly.